packages/core/streams/langchain-stream.test.ts

```diff
@@ -10,107 +29,150 @@ describe('LangchainStream', () => {
+        ',',
+        ' world',
+        '.',
+        '',
       ]);
     });
 
-    it('should send text and data', async () => {
-      const data = new experimental_StreamData();
+    describe('StreamData prototcol', () => {
+      it('should send text', async () => {
+        const data = new experimental_StreamData();
 
-      data.append({ t1: 'v1' });
+        const model = new ChatOpenAI({
+          streaming: true,
+          openAIApiKey: 'fake',
+          configuration: {
+            baseURL: server.api,
+            defaultHeaders: {
+              'x-mock-service': 'openai',
+              'x-mock-type': 'chat',
+            },
+          },
+        });
 
-      const { stream, handlers } = LangChainStream({
-        onFinal() {
-          data.close();
-        },
-        experimental_streamData: true,
+        const stream = await PromptTemplate.fromTemplate('{input}')
+          .pipe(model)
+          .pipe(new BytesOutputParser())
+          .stream(
+            { input: 'Hello' },
+            {
+              callbacks: [
+                {
+                  handleChainEnd(outputs, runId, parentRunId) {
+                    // check that main chain (without parent) is finished:
+                    if (parentRunId == null) {
+                      data.close();
+                    }
+                  },
+                },
+              ],
+            },
+          );
+
+        const response = new StreamingTextResponse(
+          stream.pipeThrough(createStreamDataTransformer(true)),
+          {},
+          data,
+        );
+
+        expect(await readAllChunks(response)).toEqual([
+          '0:""\n',
+          '0:"Hello"\n',
+          '0:","\n',
+          '0:" world"\n',
+          '0:"."\n',
+          '0:""\n',
+        ]);
+      });
+
+      it('should send text and data', async () => {
+        const data = new experimental_StreamData();
+
+        data.append({ t1: 'v1' });
+
+        const model = new ChatOpenAI({
+          streaming: true,
+          openAIApiKey: 'fake',
+          configuration: {
+            baseURL: server.api,
+            defaultHeaders: {
+              'x-mock-service': 'openai',
+              'x-mock-type': 'chat',
+            },
+          },
+        });
+
+        const stream = await PromptTemplate.fromTemplate('{input}')
+          .pipe(model)
+          .pipe(new BytesOutputParser())
+          .stream(
+            { input: 'Hello' },
+            {
+              callbacks: [
+                {
+                  handleChainEnd(outputs, runId, parentRunId) {
+                    // check that main chain (without parent) is finished:
+                    if (parentRunId == null) {
+                      data.close();
+                    }
+                  },
+                },
+              ],
+            },
+          );
+
+        const response = new StreamingTextResponse(
+          stream.pipeThrough(createStreamDataTransformer(true)),
```

Comment:
The langchain stream also needs to go through createStreamDataTransformer.

Reply:
We should move this into langchain-stream

Reply:
LangChain has two interfaces: the classic chains (what we support in `langchain-stream`) and the new langchain expression language (which is tested here and has a slightly different API). I see two options:

1) Introducing a separate LangChainExpressionLanguageStream
2) Introducing an options argument on LangChainStream for switching to LangChainExpressionLanguage support

Since the interface for working with the expression langauge is very different (see below), I'm leaning towards option 1). @MaxLeiter what are your thoughts?

### LangChain expression language right now
```ts
const data = new experimental_StreamData();

const model = new ChatOpenAI({
  //...
});

const stream = await PromptTemplate.fromTemplate('{input}')
  .pipe(model)
  .pipe(new BytesOutputParser())
  .stream(
    { input: 'Hello' },
    {
      callbacks: [
        {
          handleChainEnd(outputs, runId, parentRunId) {
            // check that main chain (without parent) is finished:
            if (parentRunId == null) {
              data.close();
            }
          },
        },
      ],
    },
  );

const response = new StreamingTextResponse(
  // stream comes from LangChain, we could e.g. wrap it
  stream.pipeThrough(createStreamDataTransformer(true)),
  {},
  data,
);
```

### LangChain classic right now
```ts
const data = new experimental_StreamData();

const model = new ChatOpenAI({
  // ...
});

const { stream, handlers } = LangChainStream({
  onFinal() {
    data.close();
  },
  experimental_streamData: true,
});

model
  .call([new HumanMessage('hello')], {}, [handlers])
  .catch(console.error);

const response = new StreamingTextResponse(
  // stream is synthetically created by the langchain callback handlers
  stream, 
  {}, 
  data
);
```